### PR TITLE
[1LP][RFR] Dockerbot collectonly step hack

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
work in progress adding the pytest options and the init args as well as the not yet tested code in the prt image commands

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d3856b321be6126e35fafd4ae32250bca7c73dfc:

`cfme/tests/configure/test_visual_infra.py`:
- :bangbang: Line 33:1: **E305** *expected 2 blank lines after class or function definition, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7476b7ba8d5919eb745ee1ac0c833caebfe0cd67:

`cfme/tests/configure/test_visual_infra.py`:
- :bangbang: Line 34:1: **E305** *expected 2 blank lines after class or function definition, found 1*

`cfme/utils/appliance/__init__.py`:
- :heavy_exclamation_mark: Line 2618:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/test_framework/sprout/client.py [hunk 1]
```diff
@@ -122,3 +122,6 @@ def provision_appliances(
 
     def destroy_pool(self, pool_id):
         self.call_method('destroy_pool', id=pool_id)
+
+    def all_availiable_versions(self):
```
Comment (mfalesni) [inline cfme/test_framework/sprout/client.py]: typo.

Comment (mfalesni) [reply cfme/test_framework/sprout/client.py]: And by the way, sprout client already has this behaviour, you are just overriding what is the exactly same behaviour :)

**Comment by jkrocil** [comment]
@ManageIQ/committers-qe this good enough? it's 1st and 2nd level reviewers but we don't need to create a separate team this way

File: scripts/dockerbot/pytestbase/setup.sh [hunk 2]
```diff
@@ -62,13 +62,25 @@ do_or_die () {
...
+}
+
 # Runs pip update - optionally can make use of wheelhouse
 run_pip_update () {
+	export PYCURL_SSL_LIBRARY=nss
     if [ -n "$WHEEL_HOST_URL" ]; then
-        run_n_log "PYCURL_SSL_LIBRARY=nss pip install --trusted-host $WHEEL_HOST -f $WHEEL_HOST_URL -Ur $CFME_REPO_DIR/requirements/frozen.txt --no-cache-dir"
-    else
-        run_n_log "PYCURL_SSL_LIBRARY=nss pip install -Ur $CFME_REPO_DIR/requirements/frozen.txt --no-cache-dir"
+        export PIP_TRUSTED_HOST="$WHEEL_HOST" PIP_FIND_LINKS="$WHEEL_HOST_URL"
```
Comment (psav) [inline scripts/dockerbot/pytestbase/setup.sh]: **REQUIRED:** Does this mean we must have a wheel host now?

Comment (RonnyPfannschmidt) [reply scripts/dockerbot/pytestbase/setup.sh]: no

File: scripts/dockerbot/pytestbase/setup.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@ on_exit () {
     log "Checking out master branch..."
     git checkout origin/master
     log "Running pip update..."
-    run_pip_update
+    (run_pip_update)  # subshell to avoid exit in failure
```
Comment (izapolsk) [inline scripts/dockerbot/pytestbase/setup.sh]: :+1:

**Pull request merged**